core/modules/qdisp/Executive.cc

```diff
@@ -499,20 +519,26 @@ void Executive::_reapRequesters(std::unique_lock<std::mutex> const&) {
     }
 }
 
-void Executive::_reportStatuses() {
-    std::lock_guard<std::mutex> lock(_statusesMutex);
-    StatusMap::const_iterator i,e;
-    for(i=_statuses.begin(), e=_statuses.end(); i != e; ++i) {
-        ExecStatus::Info info = i->second->getInfo();
-        std::ostringstream os;
-        os << ExecStatus::stateText(info.state)
-           << " " << info.stateCode;
-        if(!info.stateDesc.empty()) {
-            os << " (" << info.stateDesc << ")";
+void Executive::_updateProxyMessages() {
+    {
+        std::lock_guard<std::mutex> lock(_statusesMutex);
+        for(auto i=_statuses.begin(), e=_statuses.end(); i != e; ++i) {
+            auto const & info = i->second->getInfo();
+            std::ostringstream os;
+            os << info.state << " " << info.stateCode;
+            if(!info.stateDesc.empty()) {
+                os << " (" << info.stateDesc << ")";
+            }
+            os << " " << info.stateTime;
+            _messageStore->addMessage(info.resourceUnit.chunk(),
+                                      info.state, os.str());
+        }
+    }
+    {
+        std::lock_guard<std::mutex> lock(_errorsMutex);
+        if (_multiError.size()>0) {
```

Comment:
`if (!_multiError.empty()) {` should be ~better (or my-preferred most-readable variant `if (not _multiError.empty()) {`)